### PR TITLE
Feat cancellations n recommend

### PR DESCRIPTION
**Pull request by OutbreakSource**
New rule for the ordering. The location has to be five digits now. Otherwise, it will not go through

### CODE REVIEW

File: src/main/java/customerrelationsmanagement/ChartMaker.java [hunk 1]
```diff
@@ -33,13 +34,10 @@ public ChartMaker(Crud crud) {
...
 	}
 	
-	private boolean checkValidity(String time) {
-		
-		/* TODO: Daniel, check the validity
-		    check if time is valid
-		    check if row exists for the "time"
-		 */
-		return true;
+	private boolean checkValidity(String time)
```
Comment (ubello64) [inline src/main/java/customerrelationsmanagement/ChartMaker.java:37]: Nice to see there is a check for correct time wen doing analytics

File: src/main/java/customerrelationsmanagement/Emailer.java [hunk 2]
```diff
@@ -248,4 +283,39 @@ public void sendMail(String recipientAddress, String messageSubject,
...
+	 *  max of suggestive products
+	 * @param orderId
+	 *  to find matching recommendations against the current order
+	 *
+	 * @return a string for the message body
+	 *
+	 * @throws SQLException if a parameter of the query is wrong or
+	 * the connection is interrupted
+	 */
+	@NotNull private String recommendProducts(int limit, String orderId)
```
Comment (ubello64) [inline src/main/java/customerrelationsmanagement/Emailer.java:332]: I didn't know the addition of this feature would look so clean, great job using mySQL to accomplish this

**Review by ubello64 — APPROVED** [review]

**Comment by Adam-Prieto** [comment]
Reviewed 11/30. 
This seems to be a pretty substantive PR. However, given that 2 people worked on this, I assume that both were able to make sure that the changes accomplish their intended purpose. I also assume that these changes - mainly the imports - won't cause conflicts to other classes?

**Review by MXHJohn314 — APPROVED** [review]
Good job. The main things here were to make sure that the service can run in a continuous fashion from Main, and also make sure that the query for getting recommendations is accurate. Since we are able to always provide a recommendation (even when there is not enough data to recommend based on other transactions, we can just provide the top selling products), I think this user story is complete.

**Pull request merged**